cumulus/polkadot-parachain/src/service.rs

```diff
@@ -686,85 +683,6 @@ where
 	Ok(BasicQueue::new(verifier, Box::new(block_import), None, &spawner, registry))
 }
 
-/// Start an aura powered parachain node. Some system chains use this.
-pub async fn start_generic_aura_node(
```

Comment:
@s0me0ne-unkn0wn @seadanda I am not sure about removing this according to the Donal's [comment](https://github.com/paritytech/polkadot-sdk/issues/3944#issuecomment-2039431641):
> start_generic_aura_node and start_asset_hub_node are no longer needed for system parachains when they all shift to async backing in the next release, but if we're going for an omni-node approach then we'd still need start_generic_aura_node and a cli flag to enable async backing. I agree that this is a sensible default.

Reply:
I don't see any sense in keeping the generic collator, considering that the lookahead collator behaves exactly as the generic collator if a runtime doesn't support async backing. Let me know if I'm missing something

Reply:
At the time of writing that I was under the impression that the lookahead collator expected some additional RuntimeApi functions, if that's not the case and parachains without the tooling for async backing can be run with the lookahead collator then it makes the node code even cleaner. Nice

Reply:
@seadanda it took me quite some time as well to understand how [this guard](https://github.com/paritytech/polkadot-sdk/blob/c1063a530e46c4af0a934ca50422b69182255e60/cumulus/client/consensus/aura/src/collators/lookahead.rs#L461-L468) works indeed. Right now we're up to some live testing: paritytech/devops#3261

Reply:
OK that comment on `can_build_upon` makes sense.